realm/src/androidTest/java/io/realm/RealmQueryTest.java

```diff
@@ -101,6 +109,17 @@ public void testRealmQueryGreaterThan() {
         RealmQuery<AllTypes> query = testRealm.where(AllTypes.class).greaterThan(FIELD_FLOAT, 11.234567f);
         resultList = query.between(FIELD_LONG, 1, 20).findAll();
         assertEquals(10, resultList.size());
+
+        resultList = testRealm.where(AllTypes.class).greaterThan(FIELD_DATE, new Date(Long.MIN_VALUE)).findAll();
```

Comment:
With so many new asserts, I would move them to a new unit test. Something like `testGreaterThanDate`